src/serac/physics/base_physics.hpp

```diff
@@ -130,6 +130,16 @@ class BasePhysics {
    */
   virtual void outputState() const;
 
+  /**
+   * @brief Get an mfem::Operator that calculates the residual at the current outputState
+   */
+  virtual mfem::Vector outputResidual() = 0;
+
+  /**
+   * @brief Get an mfem::Operator that calculates the gradient of the residual at the current outputState
+   */
+  virtual mfem::Operator& outputGradient() = 0;
```

Comment:
Per our conversation yesterday, I think these should go in a `detail` or `impl` namespace with a note that they could be deprecated.

Reply:
I ended up `friend`ing the implementation. This gets rid of the implementation inheritance which is a little brittle at the moment.